src/core/Akka/Util/MurmurHash.cs

```diff
@@ -202,20 +202,20 @@ public static int StringHash(string s)
         {
             unchecked
             {
-                var sChar = s.ToCharArray();
```

Comment:
This is the biggest offending code. MurmurHash is called once for every DataEnvelope that are going to be sent as a gossip. For a system that are trying to gossip thousands of gossips, this can add up quite significantly.

Reply:
Before

|              Method |      Mean |    Error |   StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------------------- |----------:|---------:|---------:|-------:|------:|------:|----------:|
|  Murmur_string_hash |  60.87 ns | 1.240 ns | 2.389 ns | 0.0229 |     - |     - |      96 B |
| Jenkins_string_hash | 221.14 ns | 0.767 ns | 0.680 ns | 0.0153 |     - |     - |      64 B |
|  Murmur_binary_hash | 166.72 ns | 2.177 ns | 2.036 ns |      - |     - |     - |         - |
| Jenkins_binary_hash | 370.40 ns | 0.709 ns | 0.554 ns |      - |     - |     - |         - |

Reply:
``` ini

BenchmarkDotNet=v0.13.0, OS=Windows 10.0.19041.985 (2004/May2020Update/20H1)
AMD Ryzen 7 1700, 1 CPU, 16 logical and 8 physical cores
.NET SDK=5.0.203
  [Host]     : .NET Core 3.1.15 (CoreCLR 4.700.21.21202, CoreFX 4.700.21.21402), X64 RyuJIT
  DefaultJob : .NET Core 3.1.15 (CoreCLR 4.700.21.21202, CoreFX 4.700.21.21402), X64 RyuJIT


```
|              Method |      Mean |    Error |   StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------------------- |----------:|---------:|---------:|-------:|------:|------:|----------:|
|  Murmur_string_hash |  40.39 ns | 0.082 ns | 0.077 ns |      - |     - |     - |         - |
| Jenkins_string_hash | 218.54 ns | 0.467 ns | 0.390 ns | 0.0153 |     - |     - |      64 B |
|  Murmur_binary_hash | 166.06 ns | 2.224 ns | 2.080 ns |      - |     - |     - |         - |
| Jenkins_binary_hash | 372.00 ns | 2.132 ns | 1.664 ns |      - |     - |     - |         - |

Reply:
That is definitely a big improvement.